pkg/nfd-master/nfd-master.go

```diff
@@ -162,6 +162,64 @@ type nfdMaster struct {
 	config *NFDConfig
 }
 
+// customHealthServer implements grpc_health_v1.HealthServer
+type customHealthServer struct {
+	grpc_health_v1.UnimplementedHealthServer
+	k8sclient k8sclient.Interface
+}
+
+func (s *customHealthServer) SetK8sClient(client k8sclient.Interface) {
+	s.k8sclient = client
+}
+
+// Check method for customHealthServer
+func (s *customHealthServer) Check(ctx context.Context, req *grpc_health_v1.HealthCheckRequest) (*grpc_health_v1.HealthCheckResponse, error) {
+	klog.InfoS("Check request received")
+	metricServerStatus, err := s.CheckPods("kube-system", "k8s-app=metrics-server")
```

Comment:
Oh, sorry for the ambiguity. I wasn't thinking about kubernetes metrics-server but the metrics http server inside nfd-master itself that is started with `utils.CreateMetricsServer()`. We could e.g. just set a health flag to false if the goroutine started there exits.